### PR TITLE
sysbuild: support loading of CMakeCache from Zephyr projects

### PR DESCRIPTION
**Pull request by tejlmand**
This commit loads the CMakeCache of Zephyr projects.

This allows sysbuild to fetch information from Zephyr projects into sysbuild itself.

This commit is a first step in the process of sharing more knowledge between images and sysbuild, and pave the way for closer sharing of settings between images.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

### CODE REVIEW

**Comment by tejlmand** [comment]
@danieldegrasse although not completely there yet, see this as one more step in the direction of more formally share setting information between images as discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/48709#discussion_r947581332

File: share/sysbuild/cmake/modules/sysbuild_extensions.cmake [hunk 1]
```diff
@@ -2,6 +2,83 @@
...
+#
+#   sysbuild_get(my_sample_PROJECT_NAME IMAGE my_sample VAR PROJECT_NAME)
+#     will lookup PROJECT_NAME from the CMakeCache identified by `my_sample` and
+#     and return the value in the local variable `my_sample_PROJECT_NAME`.
+#
+# <variable>: variable used for returning CMake cache value. Also used as lookup
+#             variable if `VAR` is not provided.
+# IMAGE:      image name identifying the cache to use for variable lookup.
+# VAR:        name of the CMake cache variable name to lookup.
+function(sysbuild_get variable)
```
Comment (nordicjm) [inline share/sysbuild/cmake/modules/sysbuild_extensions.cmake:65]: This does not empty the input variable if the VAR variable does not exist in the target. E.g. if the input variable holds "test" and VAR is set to "BOB" which doesn't exist, the input variable will still contain "test"

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: share/sysbuild/cmake/modules/sysbuild_extensions.cmake (hunk 1 above)
Comment (tejlmand) [reply share/sysbuild/cmake/modules/sysbuild_extensions.cmake:65]: > This does not empty the input variable if the VAR variable does not exist in the target. E.g. if the input variable holds "test" and VAR is set to "BOB" which doesn't exist, the input variable will still contain "test"

That is correct.
In this case it's actually the callers responsibility to ensure the var is undefined.
A couple of reasons:
- We cannot use the general CMake approach of `<var>-NOTFOUND` because a cache var could contain the value `var-NOTFOUND` making it impossible to differentiate the two case / as to why `-NOTFOUND` was returned.
- Clearing the var using `unset(<var> PARENT_SCOPE)` would only clear the normal var, but the var could actually be a cache var, and thus still be present. And we should not clear sysbuild cache vars in such cases.
- We cannot set the var to empty string, cause the real cache var could actually be an empty string.

so basically it's the callers responsibility to ensure the var is undefined, if that is important to know.

What we can do is to print a warning if the return var is already defined with a value, and maybe even hint the caller that `sysbuild_get(<var> ... VAR <imager-variable>)` should be used instead.

File: share/sysbuild/cmake/modules/sysbuild_extensions.cmake [hunk 2]
```diff
@@ -2,6 +2,83 @@
...
+# This function will load the CMakeCache.txt file from the binary directory
+# given by the BINARY_DIR argument.
+#
+# All CMake cache variables are stored in a custom target which is identified by
+# the name given as value to the IMAGE argument.
+#
+# IMAGE:      image name identifying the cache for later sysbuild_get() lookup calls.
+# BINARY_DIR: binary directory (build dir) containing the CMakeCache.txt file to load.
+function(load_cache)
+  set(single_args APPLICATION BINARY_DIR)
```
Comment (danieldegrasse) [inline share/sysbuild/cmake/modules/sysbuild_extensions.cmake:17]: documentation here does not seem to match arguments, should `IMAGE` be `APPLICATION`?

Comment (tejlmand) [reply share/sysbuild/cmake/modules/sysbuild_extensions.cmake:17]: fixed.

I changed from APP.. --> IMAGE to have the functions better aligned, but seems I forgot to do it allover the place.
Thanks for spotting.

File: share/sysbuild/cmake/modules/sysbuild_extensions.cmake (hunk 1 above)
Comment (tejlmand) [reply share/sysbuild/cmake/modules/sysbuild_extensions.cmake:65]: Warning printed if caller is using an already defined variable.

**Review by nordicjm — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]
